### ISSUE TITLE
Missing plugin exception in obfuscated build

### ISSUE THREAD

**Issue by nemscep** [SAP/gigya-flutter-plugin#67]
Hello there! 👋 

When app is build in release (minified R8) mode, I keep getting this exception: 

`Unhandled Exception: MissingPluginException(No implementation found for method isLoggedIn on channel com.sap.gigya_flutter_plugin/methods)`

If I set minify to false, it is working as expected, but this is a "dirty" workaround since minification is a must for all android builds going live.
So, instead of disabling minification globally, a kinda better workaround I found is utilizing proguard to skip minification for only gigya sdk stuff like this: 

`-keep class com.gigya.** { *; }`

Even though proguard is an optimal workaround, this seems like a library issue which needs fixing 🤷‍♂️ 

Version: 1.0.1
Platform: Android
Flutter version: 3.10.3 

Cheers ✌️

**Comment by navaronbracke** [comment]
_TLDR for anyone that wants a workaround: You can add the above rules to a Proguard rules file in your Android project_

@nemscep You are correct about the Proguard rules file (I personally had this with other plugins before too). So we should add a Proguard rules file in the plugin. Can you open a PR that adds this?

See the code sample at https://developer.android.com/studio/projects/android-library#Considerations
and this is how `integration_test` does it: https://github.com/flutter/flutter/blob/master/packages/integration_test/android/build.gradle

So you'll have to add the `consumerProguardFiles` in the `android/build.gradle` of the plugin (not the example app!), and put the rules that you mentioned above in the specified file. For testing, you can then try to run the example app in release mode and verify that it works.

**Comment by nemscep** [comment]
> _TLDR for anyone that wants a workaround: You can add the above rules to a Proguard rules file in your Android project_
> 
> @nemscep You are correct about the Proguard rules file (I personally had this with other plugins before too). So we should add a Proguard rules file in the plugin. Can you open a PR that adds this?
> 
> See the code sample at https://developer.android.com/studio/projects/android-library#Considerations and this is how `integration_test` does it: https://github.com/flutter/flutter/blob/master/packages/integration_test/android/build.gradle
> 
> So you'll have to add the `consumerProguardFiles` in the `android/build.gradle` of the plugin (not the example app!), and put the rules that you mentioned above in the specified file. For testing, you can then try to run the example app in release mode and verify that it works.

Sure thing! I create PR for this 😉

**Comment by nemscep** [comment]
> > _TLDR for anyone that wants a workaround: You can add the above rules to a Proguard rules file in your Android project_
> > @nemscep You are correct about the Proguard rules file (I personally had this with other plugins before too). So we should add a Proguard rules file in the plugin. Can you open a PR that adds this?
> > See the code sample at https://developer.android.com/studio/projects/android-library#Considerations and this is how `integration_test` does it: https://github.com/flutter/flutter/blob/master/packages/integration_test/android/build.gradle
> > So you'll have to add the `consumerProguardFiles` in the `android/build.gradle` of the plugin (not the example app!), and put the rules that you mentioned above in the specified file. For testing, you can then try to run the example app in release mode and verify that it works.
> 
> Sure thing! I create PR for this 😉

Here it is https://github.com/SAP/gigya-flutter-plugin/pull/68

**Comment by navaronbracke** [comment]
cc @tal-mi since this was resolved, the issue can be closed.

**Comment by tal-mi** [comment]
Yeah.
